tests/e2e.sh

```diff
@@ -14,6 +14,19 @@ kind() {
     ./kind create cluster
 }
 
+prepare-e2e() {
+    # in a real OpenShift cluster, this secret is automatically created and populated
+    # based on the annotation in the dex service. For tests, we need to manually create this.
+    $KUBECTL create ns dex || true
+    $KUBECTL apply -f tests/manifests/observatorium-xyz-tls-dex.yaml
+
+    # in a real OpenShift cluster, this configmap is automatically populated with the service-ca
+    # upon creation. For the e2e tests, we need to manually populate it.
+    $KUBECTL create ns observatorium || true
+    cp environments/dev/manifests/test-ca-tls.yaml environments/dev/manifests/test-ca-tls.yaml.orig
+    cp tests/manifests/test-ca-tls.yaml environments/dev/manifests/test-ca-tls.yaml
+}
+
```

Comment:
I don't think we need this entire new function here. It should be possible to integrate with the deploy function, simliar to what we have with in the operator repository: https://github.com/observatorium/operator/blob/master/tests/e2e.sh#L17-L34

Reply:
+1, I remember facing a mismatch in the steps in some earlier iterations, but looks like it doesn't exist anymore. I removed this and it works with a single `deploy` function.